node_modules/col-assets/tests/test-bookmarklet.js

```diff
@@ -33,155 +34,97 @@ var UsersAPI = require('col-users');
-   */
-  it('is validated', function(callback) {
-    TestsUtil.getAssetLibraryClient(null, null, null, function(client, course, user) {
-      UsersTestUtil.assertGetMe(client, course, null, function(me) {
-        var bookmarkletClient = TestsUtil.getAnonymousClient();
-
-        // Too long title
-        AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, randomstring.generate(256), 'http://www.berkeley.edu/', null, 400, function() {
-
-          // Missing URL
-          AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', null, null, 400, function() {
-            // Invalid URL
-            AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'invalid url', null, 400, function() {
-              AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', '/invalidurl', null, 400, function() {
-                // Too long URL
-                AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/?q=' + randomstring.generate(229), null, 400, function() {
-
-                  // Invalid source
-                  AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/', {'source': 'invalid url'}, 400, function() {
-                    AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/', {'source': '/invalidurl'}, 400, function() {
-
-                      return callback();
-                    });
-                  });
-                });
-              });
-            });
-          });
-        });
-      });
+  it('creates links with no optional metadata', function(callback) {
+    AssetsTestUtil.assertCreateLinkBookmarklet(bookmarkletClient, testCourse, testMe.id, testMe.bookmarklet_token, 'UC Davis', 'http://www.ucdavis.edu/', null, function(asset) {
+      callback();
+    });
+  });
+  it('defaults the link title to the URL', function(callback) {
+    var url = 'http://uci.edu';
+    AssetsTestUtil.assertCreateLinkBookmarklet(bookmarkletClient, testCourse, testMe.id, testMe.bookmarklet_token, null, url, null, function (asset) {
+      assert.equal(asset.title, url);
+      callback();
+    });
+  });
+  it('creates links with optional metadata', function(callback) {
+    var opts = {
+      'description': 'University of California, Berkeley homepage',
+      'source': 'http://www.universityofcalifornia.edu/uc-system'
+    };
+    AssetsTestUtil.assertCreateLinkBookmarklet(bookmarkletClient, testCourse, testMe.id, testMe.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/', opts, function(asset) {
+      callback();
```

Comment:
Thank you, @paulkerschen . I was wondering about that convention & the explanation makes sense. I'll amend, since you're pro-merge.

While we're on the subject of conventions, I note that Mocha's docs always name their async "I-am-all-done-with-my-test" function argument "done". As a newcomer, that made sense to me as a way to distinguish it from customer-authored callback functions. Do you know why PDG uses the more generic name?